azerothcore/azerothcore-wotlk
# fix(Script/Instance) [Deadmines] Smite Slam

## Changes Proposed:
- Moves Smite Slam to be more chronological
- Resolves the issue of him constantly casting by setting a 33% health condition for the spell
- Adds the proper cooldown (6 sec)
- Removes Smite Slam from the creature_template_spell (prevents him from constantly casting it)


## Issues Addressed:
- Closes #4945



## SOURCE:

See original report.

## Tests Performed:

I tested the encounter by myself `.go c id 646` , got him to 33% and checked combatlog.


## How to Test the Changes:

Please see the wiki on how to test PR.
Enter the instance by `.go c id 646`. 
Get him to 33% hp and check combatlog (he might miss the spell, takes up to 6 seconds for him to cast because of first cooldown)

## Known Issues and TODO List:

He should not be able to miss the spell (according to wowhead) 
The mobpack around him will get pulled no matter what, there is a greater issue that causes it and there is already a seperate bugtracker about it. I am interested in solving in but I am not proficient enough at this time.

## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

Note: My first C++ PR, please have a good look.
